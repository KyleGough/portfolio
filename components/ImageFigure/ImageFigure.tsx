import { Section } from '@components/Section';
import React from 'react';

interface ImageFigureProps {
  image: {
    src: string;
    alt: string;
  };
}

export const ImageFigure: React.FC<ImageFigureProps> = ({ image }) => (
  <Section>
    <img
      className="border-2 border-white rounded-lg text-center items-center shadow mx-auto"
      src={image.src}
      alt={image.alt}
    />
    <p className="mt-4 mb-2 text-sm text-center text-light font-thin">
      {image.alt}
    </p>
  </Section>
);
