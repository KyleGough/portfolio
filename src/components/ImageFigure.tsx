import Section from './Section';

type FigureProps = {
    image: {
        src: string,
        caption: string
    }
};

export default function ImageFigure(props: FigureProps) {
    return (
        <Section>
            <img className='border-2 border-white rounded-lg text-center items-center shadow mx-auto' src={props.image.src} alt={props.image.caption} />
            <p className='mt-4 mb-2 text-sm text-center text-light font-thin'>{props.image.caption}</p>
        </Section>
    );
}
