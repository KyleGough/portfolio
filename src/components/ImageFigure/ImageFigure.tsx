import React from 'react';
import { Section } from '../Section';

interface Image {
  src: string;
  caption: string;
}

interface ImageFigureProps {
  image: Image;
}

export const ImageFigure: React.FC<ImageFigureProps> = ({ image }) => (
  <Section>
    <img
      className="border-2 border-white rounded-lg text-center items-center shadow mx-auto"
      src={image.src}
      alt={image.caption}
    />
    <p className="mt-4 mb-2 text-sm text-center text-light font-thin">
      {image.caption}
    </p>
  </Section>
);
