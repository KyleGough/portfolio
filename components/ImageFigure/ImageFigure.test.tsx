import { render, screen } from '@testing-library/react';
import React from 'react';

import { ImageFigure } from './ImageFigure';

describe('ImageFigure component', () => {
  it('renders', async () => {
    const { container } = render(
      <ImageFigure image={{ src: '/#', alt: 'Test Caption' }} />
    );
    expect(screen.getByText('Test Caption')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
