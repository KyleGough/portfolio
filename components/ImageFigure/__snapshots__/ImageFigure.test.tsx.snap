// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageFigure component renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <img
      alt="Test Caption"
      class="border-2 border-white rounded-lg text-center items-center shadow mx-auto"
      src="/#"
    />
    <p
      class="mt-4 mb-2 text-sm text-center text-light font-thin"
    >
      Test Caption
    </p>
  </div>
</div>
`;
